tests/test_source.py

```diff
@@ -79,3 +87,58 @@ def test_broken_source_config_file(self):
         s = get_source_instance_for({'provider': 'path', 'uri': SOURCE_CONFIG_ERROR_PATH})
         with pytest.raises(ValidationError):
             s.config
+
+
+class TestSourceConfigSchemaValidation(object):
+    """Testing parsing of configuration file and schema validation.
+
+    Related to class source.SourceConfig
+    """
+    def _create_source_config(self, tmpdir, yml_config):
+        tmpdir_str = str(tmpdir)
+        path = os.path.join(tmpdir_str, REPO_CONTAINER_CONFIG)
+        # store container configuration into expected file
+        with open(path, 'w') as f:
+            f.write(yml_config)
+            f.flush()
+
+        return SourceConfig(tmpdir_str)
+
+    @pytest.mark.parametrize('yml_config', [
+        # empty config
+        """\
+        """,
+
+        """\
+        platforms:
+            only: s390x
+        """,
+
+        """\
+        platforms:
+            not: s390x
+        """,
+
+        # TODO: schema probably should not allow to use both keywords
```

Comment:
Actually the schema deliberately allows this, but it should document that "only" is processed before "not". The reason is that this allows a temporary exclusion of an architecture, e.g.:
```
only: abc def ghi jkl mno
# not yet working on ghi, bug #.....
not: ghi
```

Reply:
Thanks for clarification, TODO removed.

Test for `get_platforms_in_limits` added